src/lib/transport/ip/udp_send.c

```diff
@@ -838,7 +835,15 @@ static void ci_udp_sendmsg_send(ci_netif* ni, ci_udp_state* us,
                  __FUNCTION__));
   }
 
-  ni->state->send_may_poll = 0;
+  if( ci_netif_may_poll(ni) && ipcache->status == retrrc_success ) {
+    ci_netif_state_nic_t* nsn = &ni->state->nic[ipcache->intf_i];
+    if( nsn->tx_dmaq_insert_seq - nsn->tx_dmaq_insert_seq_last_poll >
+      NI_OPTS(ni).send_poll_thresh ) {
+      nsn->tx_dmaq_insert_seq_last_poll = nsn->tx_dmaq_insert_seq;
+      ci_netif_poll(ni);
```

Comment:
Good question.  I'd better avoid this code path when running from `ci_udp_perform_deferred_socket_work()`.  It is definitely useless (because deferred works tend to go together with polling), and may be harmful (I'm not sure, but I do not want to check carefully, and it may change in future).

Onload was so much simpler when that `send_may_poll` was implemented!